src/main/scala/is/hail/variant/HTSGenotypeView.scala

```diff
@@ -393,4 +393,91 @@ class HardCallStructView(rowType: TStruct, callField: String) extends HardCallVi
       throw new DataFormatException(s"Expected call to be non-negative, but found $gt")
     gt
   }
-}
\ No newline at end of file
+}
+
+class HardcallTrioGenotypeView(rs: TStruct, callField: String) {
+  private val trioTgs = rs.fieldType(3).asInstanceOf[TArray]
+  private val trioTg = trioTgs.elementType.asInstanceOf[TStruct]
+
+  require(trioTg.hasField("proband") && trioTg.hasField("mother") && trioTg.hasField("father"))
+  private val probandIndex = trioTg.fieldIdx("proband")
+  private val motherIndex = trioTg.fieldIdx("mother")
+  private val fatherIndex = trioTg.fieldIdx("father")
+
+  private val tg = trioTg.fieldType(probandIndex).fundamentalType.asInstanceOf[TStruct]
+
+  require(tg.hasField(callField))
+  private val gtIndex = tg.fieldIdx(callField)
+  require(tg.unify(trioTg.fieldType(motherIndex).fundamentalType) && tg.unify(trioTg.fieldType(fatherIndex).fundamentalType))
+
+  private var m: MemoryBuffer = _
+  private var gsOffset: Long = _
+  private var gsLength: Int = _
+  private var gOffset: Long = _
+  private var gProbandOffset: Long = _
+  private var gMotherOffset: Long = _
+  private var gFatherOffset: Long = _
+
+  var gIsDefined: Boolean = _
+
+  def setRegion(mb: MemoryBuffer, offset: Long) {
+    this.m = mb
+    gsOffset = rs.loadField(m, offset, 3)
+    gsLength = trioTgs.loadLength(m, gsOffset)
+  }
+
+  def setRegion(rv: RegionValue) {
+    setRegion(rv.region, rv.offset)
+  }
+
+  def setGenotype(idx: Int) {
+    require(idx >= 0 && idx < gsLength)
+    gIsDefined = trioTgs.isElementDefined(m, gsOffset, idx)
```

Comment:
this is really more like `trioIsDefined`, right?

Reply:
This is "the trio-genotype is defined for trio idx at this variant". This is like the other views, but with trios in place of samples, and the other views call this `gIsDefined`, not `sampleIsDefined`. I extended the naming convention of the other views to let `g` refer to the compound trio-genotype, with `gProband`, `gMother`, and `gFather` refering to the individual genotypes. But I can change it if you think it's confusing here.

Reply:
Ah, OK, I see the logic. I do think it's a bit odd, but let's stick with it.